### PR TITLE
create static problem_data_schema

### PR DESCRIPTION
**Pull request by FelixBenning**
static problem_data schema

### Unsolved issue:
add file moving functionality to `make_schemas`. This also requires finding a way to distribute this static json file with the source which is nontrivial and requires
- [ ]  moving the file to the build tree as a first step, at which point you have found a solution for the gui problem in #224.
- [ ] bundle the file not only with the gui (which is likely easier because gooey has functionality for that) but also with the command line program which is again non-trivial and makes the install more complicated than simply moving the executable grazer file into the bin folder
- [ ] write a copying mechanism into `make_schemas` copying the bundled json into the `schemas` folder together with the other schemas. Since the bundling mechanism is different on different platforms (e.g. together with the binary in the grazer folder on windows, separate from the binary which is located in a `bin` directory on UNIX) this is probably again non-trivial since the retrieval mechanism must take into account the OS

### Workaround:
Keep the schema in the docs and let people copy it manually (not pretty)

### Workaround2:

hardcode the json as a raw string into cpp files (possibly use cmake environment variable replacement in files)

### CODE REVIEW

**Comment by eike-fokken** [comment]
This actually is not what I had in mind. I meant an addition to grazer that creates the schema just like we do for the other schemas.

I looked into file copying during the install step and it seems to be impossible in a platform-agnostic way.

**Comment by FelixBenning** [comment]
implemented workaround2 in b128004 (hardcoded the json into the schema generation)

**Comment by eike-fokken** [comment]
Please have a look at my latest commit. I stumbled upon another problem, only tangentially related:
When I regenerate the schemas, they do not take defaults into account, as we discussed in #222 
Could you have a look into that?

Apart from that, I'd merge this branch.

File: src/Auxillary/json_schema/schema_generation.cpp [hunk 1]
```diff
@@ -52,9 +52,12 @@ namespace Aux::schema {
     return 0;
   }
 
+  inline static json get_problem_data_schema();
+
   void make_schemas(Componentfactory const &factory, path schema_dir) {
 
     std::map<std::string, json> schemas{
+        {"problem_data", get_problem_data_schema()},
```
Comment (FelixBenning) [inline src/Auxillary/json_schema/schema_generation.cpp:60]: the following lines, e.g.
```cpp
{"topology",
 factory.get_topology_schema(
             /*allow_required_defaults=*/false, /*include_external=*/false)},
```
should make sure that the "required" property is removed from the schemas - I am not sure why this is not working as intended. 
This function calls `component->get_topology_schema(allow_required_defaults)` defined in the templates in `Componentfactory.hpp` and these individual components call `remove_defaults_from_required` from `make_schema.cpp` if `allow_required_defaults` is false.

`remove_defaults_from_required` should remove all components with defaults.

Comment (FelixBenning) [reply src/Auxillary/json_schema/schema_generation.cpp:60]: e.g. 

```cpp
if (opt_boundary_schema.has_value()) {
          auto boundary_schema = opt_boundary_schema.value();
          Aux::schema::add_defaults(
              boundary_schema,
              this->defaults.value("boundary_values", R"({})"_json));
          if (not allow_required_defaults) {
            Aux::schema::remove_defaults_from_required(boundary_schema);
          }
        }
```
`Aux::schema::add_defaults` is tested in `schemaCreationTest.cpp` so I am fairly certain this is working so I would probably set debug points in `remove_defaults_from_required`

Comment (eike-fokken) [reply src/Auxillary/json_schema/schema_generation.cpp:60]: Does it work on your end?

Comment (eike-fokken) [reply src/Auxillary/json_schema/schema_generation.cpp:60]: Fixed: You changed a copy of the required_list. I changed it to a reference. Will merge tomorrow.

**Pull request merged**